src/raftstore/coprocessor/region_info_accessor.rs

```diff
@@ -98,9 +98,8 @@ impl Display for RegionInfoQuery {
 
 /// `RegionEventListener` implements observer traits. It simply send the events that we are interested in
 /// through the `scheduler`.
-#[derive(Clone)]
 struct RegionEventListener {
-    scheduler: Scheduler<RegionInfoQuery>,
+    scheduler: Mutex<Scheduler<RegionInfoQuery>>,
```

Comment:
I don't like Mutex here, IMO, Scheduler sending should be thread-safe.

Reply:
I think it's okay, the scheduler, like a sender, should be `Send + Clone`. However, I wonder if it could be a bottleneck, @rleungx could you show the performance of a rust's mutex?